### PR TITLE
feat: Consolidate internal logging configuration

### PR DESCRIPTION
**Pull request by robinmacharg**
## Goal

Consolidate internal logging configuration to only use the `BSG_LOG_LEVEL` preprocessor macro.

## Design

The smallest change possible: an import and a variable change.  Additional first-pass documentation.

## Changeset

```
BSG_KSLogger.h
BugsnagLogger.h
```

## Tests

Manual testing with the example apps, repeated running of `pod install`.  How to automate testing in an appropriate amount of time was not obvious.

### Outstanding Questions

- Platform Documentation could do with being improved (but this can be done as a separate task; raised internally)
- Additional docs (e.g. Carthage, Manual) are probably required for completeness (ditto)
- While this has been tested to a reasonable extent, the #ifdef macros are a little soupy; are there any logging edge cases I've missed?

- This pull request is ready for:
  - [X] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [X] Final review

- [ ] The correct target branch has been selected (`master` for fixes, `next` for
  features)
- [ ] Consistency across platforms for structures or concepts added or modified
- [X] Consistency between the changeset and the goal stated above
- [X] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [X] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [X] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: Source/BugsnagLogger.h [hunk 1]
```diff
@@ -1,12 +1,38 @@
+/**
+ * Higher-level user-accessible BugsnaglLogging configuration.  Controls how
```
Comment (fractalwrench) [inline Source/BugsnagLogger.h]: BugsnaglLogging -> Bugsnag Logging

File: Source/BugsnagLogger.h [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
+ *         rep.pods_project.targets.each do |target|
+ *             if target.name == "Bugsnag"
+ *                 target.build_configurations.each do |config|
+ *                     config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)', 'BSG_LOG_LEVEL=BSG_LOGLEVEL_INFO']
+ *                 end
+ *             end
+ *         end
+ *     end
+ *
+ * Change the value of `BSG_LOG_LEVEL` to one of the levels given below and run `pod install`.
```
Comment (fractalwrench) [inline Source/BugsnagLogger.h]: Is it possible to alter the log level outside of using Cocoapods? If so it'd be good to document how to achieve this in a package-manager agnostic way.

Comment (robinmacharg) [reply Source/BugsnagLogger.h]: Agreed, as noted in the PR. An internal ticket has been raised to address this. While the mechanism (`BSG_LOG_LEVEL`) __is__ applicable across installation methods (Cocoapods, Carthage and Manually), documenting the specifics for all methods of library integration is left for a later date.

Comment (fractalwrench) [reply Source/BugsnagLogger.h]: As long as a separate task has been raised and it will be addressed as part of the Notifier Spec project I'm happy to leave this for now.
